From 3fc8585cf76022dba7496627074d42af88c30718 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Sat, 23 Jun 2018 15:16:55 +0200
Subject: [PATCH 32/33] MIPS: ath79: sanitize symbols

We no longer need to select which SoCs are supported as the whole arch
code is always built. So lets drop all the SoC symbols

Signed-off-by: John Crispin <john@phrozen.org>
---
 arch/mips/Kconfig       |  2 ++
 arch/mips/ath79/Kconfig | 44 +++++---------------------------------------
 arch/mips/pci/Makefile  |  2 +-
 3 files changed, 8 insertions(+), 40 deletions(-)

--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -256,6 +256,8 @@ config ATH79
 	select SYS_SUPPORTS_BIG_ENDIAN
 	select SYS_SUPPORTS_MIPS16
 	select SYS_SUPPORTS_ZBOOT_UART_PROM
+	select HAVE_PCI
+	select USB_ARCH_HAS_EHCI
 	select USE_OF
 	select USB_EHCI_ROOT_HUB_TT if USB_EHCI_HCD_PLATFORM
 	help
--- a/arch/mips/ath79/Kconfig
+++ b/arch/mips/ath79/Kconfig
@@ -1,48 +1,14 @@
 # SPDX-License-Identifier: GPL-2.0
 if ATH79
 
-config SOC_AR71XX
-	select HAVE_PCI
-	def_bool n
-
-config SOC_AR724X
-	select HAVE_PCI
-	select PCI_AR724X if PCI
-	def_bool n
-
-config SOC_AR913X
-	def_bool n
-
-config SOC_AR933X
-	def_bool n
-
-config SOC_AR934X
-	select HAVE_PCI
-	select PCI_AR724X if PCI
-	def_bool n
-
-config SOC_QCA955X
-	select HAVE_PCI
-	select PCI_AR724X if PCI
+config PCI_AR71XX
+	bool "PCI support for AR7100 type SoCs"
+	depends on PCI
 	def_bool n
 
 config PCI_AR724X
-	def_bool n
-
-config ATH79_DEV_GPIO_BUTTONS
-	def_bool n
-
-config ATH79_DEV_LEDS_GPIO
-	def_bool n
-
-config ATH79_DEV_SPI
-	def_bool n
-
-config ATH79_DEV_USB
-	def_bool n
-
-config ATH79_DEV_WMAC
-	depends on (SOC_AR913X || SOC_AR933X || SOC_AR934X || SOC_QCA955X)
+	bool "PCI support for AR724x type SoCs"
+	depends on PCI
 	def_bool n
 
 endif
--- a/arch/mips/pci/Makefile
+++ b/arch/mips/pci/Makefile
@@ -21,7 +21,7 @@ obj-$(CONFIG_BCM63XX)		+= pci-bcm63xx.o
 					ops-bcm63xx.o
 obj-$(CONFIG_MIPS_ALCHEMY)	+= pci-alchemy.o
 obj-$(CONFIG_PCI_AR2315)	+= pci-ar2315.o
-obj-$(CONFIG_SOC_AR71XX)	+= pci-ar71xx.o
+obj-$(CONFIG_PCI_AR71XX)	+= pci-ar71xx.o
 obj-$(CONFIG_PCI_AR724X)	+= pci-ar724x.o
 obj-$(CONFIG_PCI_XTALK_BRIDGE)	+= pci-xtalk-bridge.o
 #
